.github/workflows/cicd-main.yml

```diff
@@ -5520,8 +5537,42 @@ jobs:
 
           MESSAGE=$(echo $BASE_MESSAGE | jq -c --argjson summary "$SUMMARY" '.blocks += $summary')
 
-          curl -X POST -H "Content-type: application/json" --data "$MESSAGE" ${{ secrets.SLACK_WEBHOOK }}
+          curl -X POST -H "Content-type: application/json" --data "$MESSAGE" $SLACK_WEBHOOK
 
-      - if: ${{ always() && steps.pipeline-conclusion.outputs.SUCCESS == 'false' }}
+      - name: "Pipeline not successful and not cancelled: Write summary"
+        if: ${{ always() && steps.pipeline-conclusion.outputs.FAILED == 'true' && env.SLACK_WEBHOOK != '' }}
```

Comment:
What's the functionality of this block compared to block that starts at 5456? The two conditionals look the same

Reply:
Good point, it grew a bit organically into separate blocks. But especially in this case, merging them into a single would make sense